les/handler.go

```diff
@@ -1246,5 +1282,7 @@ func (d *downloaderPeerNotify) registerPeer(p *peer) {
 
 func (d *downloaderPeerNotify) unregisterPeer(p *peer) {
 	pm := (*ProtocolManager)(d)
-	pm.downloader.UnregisterPeer(p.id)
+	if pm.ulc == nil || p.isTrusted {
```

Comment:
Are you sure this is correct? Disconnect if trusted? Maybe you wanted to keep it if it is trusted:
if pm.ulc == nil || !p.isTrusted {